src/test/java/core/basesyntax/service/impl/ProduceMapReportServiceTest.java

```diff
@@ -0,0 +1,89 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.service.Producer;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class ProduceMapReportServiceTest {
+    private static final String VALID_DATA = "banana,152" + System.lineSeparator() + "apple,90";
+    private static final String EMPTY_DATA = "";
+    private static final String ZERO_QUANTITY_DATA = "banana,0";
+    private static final String SINGLE_FRUIT_DATA = "banana,5";
+    private static final String LARGE_QUANTITY_DATA = "banana,1000"
+            + System.lineSeparator()
+            + "apple,2000";
+    private static Producer produceMapReportService;
+    private static Map<String, Integer> map;
+    private String expected;
+
+    @BeforeAll
+    public static void setUp() {
+        produceMapReportService = new ProduceMapReportService();
+        map = new LinkedHashMap<>();
+    }
+
+    @AfterEach
+    public void afterEach() {
+        map.clear();
+    }
+
+    @Test
+    public void produceMapReportService_ValidData_Ok() {
```

Comment:
there is no need to include tested class name in each test name, please remove it everywhere